### ISSUE TITLE
Spritemap animation definitions should use time-per-frame instead of frames-per-second

### ISSUE THREAD

**Issue by Draknek** [useflashpunk/FlashPunk#7]
If you use time-per-frame, it makes sense to extend it to have per-frame timing (e.g. first frame is shown for 0.5s, second frame for 0.1s, third frame for 0.5s). Doing anything like that is just confusing with fps values.

Time-per-frame also makes a lot more sense when the units of time are in frames rather than seconds ("show for 4 frames" is much more intuitive than "animate at 0.25 frames-per-frame").

Unfortunately I don't know how we can nicely change this without breaking existing code. :( That is more important than changing this.

**Comment by noonat** [comment]
You could make Spritemap.add()'s frameRate parameter an untyped value. If you pass a number, treat it as frames per second. If you pass an Array, treat it as time per frame. This would require you to always specify times for each frame individually, though.

You could also do the same thing to the frames parameter instead: Array means normal behavior, Object means you can specify time per frame. e.g.:

```
spritemap.add("foo", [1, 2, 3], 10);
spritemap.add("foo", {frames: [1, 2, 3], times: 0.1});
spritemap.add("foo", {
    frames: [1, 2, 3],
    times: [0.5, 0.1, 0.1]
});
```

**Comment by Draknek** [comment]
Don't like option 2 there. For option 1, could make it so that if the array has only one element, that value is used for all frames.

Don't know if the number vs. array behaviour would be a bit too magical (read: unintuitive) though.

Third option: add a boolean property to switch between fps or frame time. I'd personally find this very annoying though.

Fourth option: have a differently named add method. Very inelegant.

My preference for personal use would definitely be option 1, but I worry that to most users the distinction between these would not be at all obvious:

```
spritemap.add("foo", [1, 2, 3], 10);
spritemap.add("foo", [1, 2, 3], [0.1]);
```

**Comment by noonat** [comment]
Yeah, obviousness was the main reason I suggested the second option. Your array option definitely works, though you have a point about the intuitiveness of it. Although, all it takes is one peek at the docs and it's understandable forever, and the ease of use for day to day FP devs is nice. 

Do you imagine people will want to be switching back and forth between the styles in multiple add() calls on the same Spritemap? You could always make it an object level option.

**Comment by BlckKnght** [comment]
I suggest adding a static Boolean value to the SpriteMap class that would invert the interpretation of the timing parameter.

That way a programmer could set it one way for backwards compatibility in existing projects, or use the other more logical setting for projects started fresh in a new version of FlashPunk. Perhaps leave the current behavior as the default for the next release, but swap it in the release after that. Eventually -- in FP v2.0, perhaps -- you could drop the flag and just go with the time-based approach for everyone.
